addons/sale/models/account_move.py

```diff
@@ -154,6 +154,44 @@ def _is_downpayment(self):
         self.ensure_one()
         return self.line_ids.sale_line_ids and all(sale_line.is_downpayment for sale_line in self.line_ids.sale_line_ids) or False
 
+    def _get_amount_from_sales_order(self, order):
+        """
+        Consider all lines on any invoice in self that stem from the sales order `order`.
+        This function returns the sum of the totals of all those lines.
+        Note that this amount may be bigger than `order.amount_total`.
+        """
+        order_amount = 0
+        for invoice in self:
+            prices = sum(invoice.line_ids.filtered(lambda x: order in x.sale_line_ids.order_id).mapped('price_total'))
+            order_amount += invoice.currency_id._convert(
+                prices * -invoice.direction_sign,
+                order.currency_id,
+                invoice.company_id,
+                invoice.date,
+            )
+        return order_amount
+
+    def _get_partner_credit_warning_exclude_amount(self):
+        # EXTENDS module 'account'
+        # Consider the warning on a draft invoice created from a sales order.
+        # After confirming the invoice the (partial) amount (on the invoice)
+        # stemming from sales orders will be substracted from the credit_to_invoice.
+        # This will reduce the total credit of the partner.
+        # The computation should reflect the change of credit_to_invoice from 'res.partner'.
+        # (see _compute_credit_to_invoice and _compute_amount_to_invoice from 'sale.order' )
+        exclude_amount = super()._get_partner_credit_warning_exclude_amount()
+        for order in self.line_ids.sale_line_ids.order_id:
+            order_amount = min(self._get_amount_from_sales_order(order), order.amount_to_invoice)
+            order_amount_company = order.currency_id._convert(
+                # TODO: master (17.3); `max` not necessary after changes to _compute_amount_to_invoice
+                max(order_amount, 0),
+                self.company_id.currency_id,
+                self.company_id,
```

Comment:
```suggestion
                order.company_id.currency_id,
                order.company_id,
```

Reply:
I ensure that all invoices belong to a single company and convert to that company's currency.
Wdyt?

Reply:
As discussed: Changed to `ensure_one` on the move instead